rcl/CMakeLists.txt

```diff
@@ -42,6 +43,27 @@ set(${PROJECT_NAME}_sources
   src/rcl/wait.c
 )
 
+# "watch" template for changes
+configure_file(
+  "resource/logging_macros.h.em"
+  "logging_macros.h.em.watch"
+  COPYONLY)
+# generate header with logging macros
+set(python_code
+  "import em"
+  "em.invoke(['-o', 'include/rcl/logging_macros.h', '${CMAKE_CURRENT_SOURCE_DIR}/resource/logging_macros.h.em'])")
+string(REPLACE ";" "$<SEMICOLON>" python_code "${python_code}")
+add_custom_command(OUTPUT include/rcl/logging_macros.h
+  COMMAND ${CMAKE_COMMAND} -E make_directory "include/rcl"
+  COMMAND ${PYTHON_EXECUTABLE} ARGS -c "${python_code}"
+  DEPENDS "${CMAKE_CURRENT_BINARY_DIR}/logging_macros.h.em.watch"
+  COMMENT "Expanding logging_macros.h.em"
+  VERBATIM
+)
+list(APPEND ${PROJECT_NAME}_sources
+  include/rcl/logging_macros.h)
```

Comment:
Since the file is generated this makes sure the generation happens before the library is being compiled.